lib/rules/constructor-super.js

```diff
@@ -398,41 +382,34 @@ module.exports = {
                 }
 
                 // Reports if needed.
-                if (funcInfo.hasExtends) {
-                    const segments = funcInfo.currentSegments;
-                    let duplicate = false;
-                    let info = null;
+                const segments = funcInfo.currentSegments;
+                let duplicate = false;
+                let info = null;
 
-                    for (const segment of segments) {
+                for (const segment of segments) {
 
-                        if (segment.reachable) {
-                            info = segInfoMap[segment.id];
+                    if (segment.reachable) {
+                        info = segInfoMap[segment.id];
 
-                            duplicate = duplicate || info.calledInSomePaths;
-                            info.calledInSomePaths = info.calledInEveryPaths = true;
-                        }
+                        duplicate = duplicate || info.calledInSomePaths;
+                        info.calledInSomePaths = info.calledInEveryPaths = true;
                     }
+                }
 
-                    if (info) {
-                        if (duplicate) {
-                            context.report({
-                                messageId: "duplicate",
-                                node
-                            });
-                        } else if (!funcInfo.superIsConstructor) {
-                            context.report({
-                                messageId: "badSuper",
-                                node
-                            });
-                        } else {
-                            info.validNodes.push(node);
-                        }
+                if (info) {
+                    if (duplicate) {
+                        context.report({
+                            messageId: "duplicate",
+                            node
+                        });
+                    } else if (!funcInfo.superIsConstructor) {
+                        context.report({
+                            messageId: "badSuper",
+                            node
+                        });
+                    } else {
+                        info.validNodes.push(node);
                     }
-                } else if (isAnySegmentReachable(funcInfo.currentSegments)) {
-                    context.report({
-                        messageId: "unexpected",
-                        node
-                    });
                 }
```

Comment:
This was intended to report `super()` calls in classes that don't have `extends`. But such calls already cause parsing errors so that's not something a rule can catch.